azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/SnapshotImpl.java

```diff
@@ -247,7 +252,8 @@ public SnapshotImpl withDataFromVhd(String vhdUrl) {
                 .withCreationData(new CreationData())
                 .creationData()
                 .withCreateOption(DiskCreateOption.IMPORT)
-                .withSourceUri(vhdUrl);
+                .withSourceUri(vhdUrl)
+                .withStorageAccountId(buildStorageAccountId(vhdUrl));
```

Comment:
Just checking, if user call another `withStorageAccountId`, it will override this, right?

Since as javadoc, the actual storage account could in a different subscription.

Reply:
This is the enhancement we should plan. In that case, we might have new methods like `withDataFromVhd(String vhdUrl, string storageAccountId)`.

Reply:
I think you need to add this.